.github/workflows/build.yml

```diff
@@ -12,16 +12,32 @@ on:
 jobs:
   mingw:
     runs-on: ubuntu-22.04
+    env:
+      CCACHE_BASEDIR: ${{ github.workspace }}
+      CCACHE_DIR: ${{ github.workspace }}/.ccache
+      CCACHE_MAXSIZE: 500M
     strategy:
+      fail-fast: false
       matrix:
         target: [i686-w64-mingw32, x86_64-w64-mingw32]
     steps:
       - uses: actions/checkout@v3
 
+      - name: Get time
+        id: get_time
+        run: echo "timestamp=`date +%s%N`" >> $GITHUB_OUTPUT
+
+      - uses: actions/cache@v3
+        with:
+          path: ${{ env.CCACHE_DIR }}
+          key: ${{ matrix.target }}-${{ steps.get_time.outputs.timestamp }}
+          restore-keys: ${{ matrix.target }}-
+
       - name: Install dependencies
         run: |
+          sudo dpkg --add-architecture i386
           sudo apt-get update
-          sudo apt-get install -y autoconf automake pkg-config g++-mingw-w64 gcc-multilib python3-pip ninja-build nasm
+          sudo apt-get install -y autoconf automake pkg-config g++-mingw-w64 gcc-multilib python3-pip ninja-build nasm wine wine32 wine64 ccache
```

Comment:
presence of wine sometimes messes up autoconf since it thinks that being able to run the binary means we're not crosscompiling
(fyi)

Reply:
If it becomes a problem, we can think about solutions. I find it handy to test PE binary and meson also do some sanity checks when cross compiling through wine. We could even do `meson test` after little bit of copying dlls